src/USER-HMA/compute_hma.cpp

```diff
@@ -0,0 +1,473 @@
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include <mpi.h>
+#include <math.h>
+#include <string.h>
+#include "compute_hma.h"
+#include "atom.h"
+#include "update.h"
+#include "force.h"
+#include "pair.h"
+#include "bond.h"
+#include "angle.h"
+#include "dihedral.h"
+#include "improper.h"
+#include "kspace.h"
+#include "group.h"
+#include "domain.h"
+#include "modify.h"
+#include "fix.h"
+#include "fix_store.h"
+#include "memory.h"
+#include "error.h"
+#include "comm.h"
+#include "neighbor.h"
+#include "neigh_request.h"
+#include "neigh_list.h"
+
+#include <vector>
+
+
+using namespace LAMMPS_NS;
+
+/* ---------------------------------------------------------------------- */
+
+ComputeHMA::ComputeHMA(LAMMPS *lmp, int narg, char **arg) :
+  Compute(lmp, narg, arg), id_temp(NULL), deltaR(NULL)
+{
+  if (narg < 4) error->all(FLERR,"Illegal compute hma command");                    
+  if (igroup) error->all(FLERR,"Compute hma must use group all");
+  if (strcmp(arg[3],"NULL") == 0) {error->all(FLERR,"fix ID specifying the set temperature of canonical simulation is required");}       
+  else {
+    int n = strlen(arg[3]) + 1;                  
+    id_temp = new char[n];   
+    strcpy(id_temp,arg[3]);                         
+  }
+ 
+  create_attribute = 1;                  
+  extscalar = 1;                         
+  timeflag = 1;                         
+
+  // (from compute displace/atom) create a new fix STORE style  
+  // our new fix's id (id_fix)= compute-ID + COMPUTE_STORE 
+  // our new fix's group = same as compute group
+
+  int n = strlen(id) + strlen("_COMPUTE_STORE") + 1;
+  id_fix = new char[n];
+  strcpy(id_fix,id);
+  strcat(id_fix,"_COMPUTE_STORE");
+
+  char **newarg = new char*[6];     
+  newarg[0] = id_fix;
+  newarg[1] = group->names[igroup];
+  newarg[2] = (char *) "STORE";              
+  newarg[3] = (char *) "peratom";
+  newarg[4] = (char *) "1";
+  newarg[5] = (char *) "3";
+  modify->add_fix(6,newarg);        
+  fix = (FixStore *) modify->fix[modify->nfix-1];    
+ 
+  delete [] newarg;                               
+
+  // calculate xu,yu,zu for fix store array
+  // skip if reset from restart file
+
+  if (fix->restart_reset) fix->restart_reset = 0;   
+  else {
+    double **xoriginal = fix->astore;            
+    double **x = atom->x;
+    imageint *image = atom->image;
+    int nlocal = atom->nlocal;
+
+    for (int i = 0; i < nlocal; i++)
+      domain->unmap(x[i],image[i],xoriginal[i]);          
+  }
+
+  vector_flag = 1;
+  extvector = -1;
+  comm_forward = 0; // 3 if 2nd derivative needed
+
+  computeU = computeP = computeCv = -1;
+  returnAnharmonic = 0;
+  std::vector<int> extvec;
+  for (int iarg=4; iarg<narg; iarg++) {
+    if (!strcasecmp(arg[iarg], "u")) {
```

Comment:
`stracasecmp()` is a POSIX function and thus not available on all platforms. in LAMMPS we prefer using (abbreviated) words as keywords and have them all lower case. so switching to the (C89 compatible) `strcmp()` function may be preferred